src/runtime/composables/useSlideover.ts

```diff
@@ -1,63 +1,84 @@
 import type { ComponentProps } from '../types/component'
-import type { Slideover, SlideoverState } from '../types/slideover'
+import type { Slideover, SlideoverInstance } from '../types/slideover'
 
-export const slidOverInjectionKey: InjectionKey<ShallowRef<SlideoverState>> = Symbol('nuxt-ui.slideover')
+export const slidOverInjectionKey: InjectionKey<ShallowRef<SlideoverInstance[]>> = Symbol('nuxt-ui.slideover')
 
 function _useSlideover () {
-  const slideoverState = inject(slidOverInjectionKey)
-
-  const isOpen = ref(false)
+  const slideoverInstances = inject(slidOverInjectionKey)
 
   function open<T extends Component> (component: T, props?: Slideover & ComponentProps<T>) {
-    if (!slideoverState) {
+    if (!slideoverInstances) {
       throw new Error('useSlideover() is called without provider')
     }
+    
+    const instance = createInstance(component, props)
 
-    slideoverState.value = {
-      component,
-      props: props ?? {}
-    }
+    slideoverInstances.value = [
+      ...slideoverInstances.value,
+      instance
+    ]
 
-    isOpen.value = true
+    return instance
   }
 
-  async function close () {
-    if (!slideoverState) return
+  async function close (id: number) {
+    const slideoverInstance = slideoverInstances.value.find((slideover) => slideover.id === id)
+    
+    if (!slideoverInstance) return
 
-    isOpen.value = false
+    slideoverInstance.isOpen = false
+    
+    slideoverInstances.value = [
+      ...slideoverInstances.value.filter((slideover) => slideover.id !== id),
+      {
+        ...slideoverInstance,        
+        isOpen: false      
+      }
+    ]
   }
 
-  function reset () {
-    slideoverState.value = {
-      component: 'div',
-      props: {}
-    }
+  function reset (id: number) {     
+    slideoverInstances.value = slideoverInstances.value.filter((slideover) => slideover.id !== id)    
   }
 
   /**
    * Allows updating the slideover props
    */
-  function patch<T extends Component = {}> (props: Partial<Slideover & ComponentProps<T>>) {
-    if (!slideoverState) return
+  function patch<T extends Component = {}> (id: number, props: Partial<Slideover & ComponentProps<T>>) {
+    const slideoverInstance = slideoverInstances.value.find((slideover) => slideover.id === id)
+
+    if (!slideoverInstance) return
 
-    slideoverState.value = {
-      ...slideoverState.value,
-      props: {
-        ...slideoverState.value.props,
+    slideoverInstances.value = [
+      ...slideoverInstances.value.filter((slideover) => slideover.id !== id),
+      {
+        ...slideoverInstance,
         ...props
       }
-    }
+    ]
   }
 
+  function createInstance<T extends Component> (component: T, props?: Slideover & ComponentProps<T>): SlideoverInstance {
+    // Random short id
+    const id = Math.floor(Math.random() * 1000000)
+
+    return {      
+        id,
+        isOpen: true,
+        component,
+        props: props ?? {},
+        close: () => close(id)        
```

Comment:
Would it make sense to add `reset` on the instance?

It feels like `reset` is primarily used internally, and it would be hardly helpful in user land

Reply:
I don't think it should

Reply:
I don't think it should